Symfony/CS/Tests/Tokenizer/TokenTest.php

```diff
@@ -346,4 +346,30 @@ public function provideEqualsAny()
             array(false, array(array(T_VARIABLE, 'junction'), array(T_FUNCTION, 'junction')), false),
         );
     }
+
+    /**
+     * @dataProvider provideIsKeyCaseSensitive
+     */
+    public function testIsKeyCaseSensitive($isKeyCaseSensitive, $caseSensitive, $key)
+    {
+        $this->assertSame($isKeyCaseSensitive, Token::isKeyCaseSensitive($caseSensitive, $key));
+    }
+
+    public function provideIsKeyCaseSensitive()
+    {
+        return array(
+            array(true, true, 0),
+            array(true, true, 1),
+            array(true, array(), 0),
+            array(true, array(true), 0),
+            array(true, array(false, true), 1),
+            array(true, array(false, true, false), 1),
+
+            array(false, false, 10),
+            array(false, array(false), 0),
+            array(false, array(true, false), 1),
+            array(false, array(true, false, true), 1),
+            array(false, array(1 => false), 1),
```

Comment:
Please add a test when key is outside of given array.
The `Token::isKeyCaseSensitive` handle it, so it would be cool to cover that case too!